app/services/order_cycle_form.rb

```diff
@@ -23,29 +24,59 @@ def save
       order_cycle.schedule_ids = schedule_ids
       order_cycle.save!
       apply_exchange_changes
+      attach_selected_shipping_methods
       sync_subscriptions
       true
     end
-  rescue ActiveRecord::RecordInvalid
+  rescue ActiveRecord::RecordInvalid => e
+    add_exception_to_order_cycle_errors(e)
     false
   end
 
   private
 
   attr_accessor :order_cycle, :order_cycle_params, :user, :permissions
 
+  def add_exception_to_order_cycle_errors(exception)
+    error = exception.message.split(":").last.strip
+    order_cycle.errors.add(:base, error) if !order_cycle.errors.to_a.include?(error)
```

Comment:
Ruby developers don't like `!`. :wink: 

```suggestion
    order_cycle.errors.add(:base, error) if order_cycle.errors.to_a.exclude?(error)
```